client/src/newsletter/PermissionCreator.tsx

```diff
@@ -81,51 +94,56 @@ const INITIAL_STATE = {
     description: '',
     consentText: '',
   },
-  alert: {
-    status: null,
-    msg: '',
+  errors: {
+    name: '',
+    frequency: '',
+    description: '',
+    consentText: '',
   },
 } as InitialState;
 
 export const PermissionCreator = (): JSX.Element => {
   const [formValues, setFormValues] = React.useState<FormValues>(
     INITIAL_STATE.values,
   );
-  const [alert, setAlert] = React.useState<AlertsState>(INITIAL_STATE.alert);
+  const [formErrors, setFormErrors] = React.useState<FormErrors>(
+    INITIAL_STATE.errors,
+  );
+  const [alert, setAlert] = React.useState<string | null>(null);
   const [isPending, setIsPending] = React.useState<boolean>(false);
   const newsletterSdk = useNewsletterSdk();
   const history = useHistory();
 
-  React.useEffect(() => {
-    let timeoutId: NodeJS.Timeout | undefined;
-    if (alert.status === 'success') {
-      timeoutId = setTimeout(() => {
-        history.goBack();
-      }, 2000);
-    }
-
-    return () => {
-      if (timeoutId) {
-        clearTimeout(timeoutId);
-      }
-    };
-  }, [alert.status]);
+  const handleSwitchRoute = (action: 'cancel' | 'success') => {
+    history.push({
+      pathname: '/newsletters',
+      state: action === 'success' ? { permissionCreated: 'success' } : {},
+    });
+  };
 
   const createNewPermission = async (event: React.FormEvent): Promise<void> => {
     try {
       event.preventDefault();
       setIsPending(true);
+      const { isValidate, newErrorState } = validateForm(formValues);
 
-      if (!validateForm(formValues)) {
-        throw new Error('Fill form correctly!');
+      if (isValidate) {
+        const newPermissionDto = createPermissionDto(formValues);
+        await newsletterSdk.permissions.admin.create(newPermissionDto);
+      } else {
+        setFormErrors((formErrors) => ({
+          ...formErrors,
+          ...newErrorState,
+        }));
+        throw new Error('validation');
       }
```

Comment:
invert to make it more readable

```suggestion
      if (!isValid) {
         setFormErrors((formErrors) => ({
          ...formErrors,
          ...newErrorState,
        }));
        throw new Error('validation');
      }
      
       const newPermissionDto = createPermissionDto(formValues);
       await newsletterSdk.permissions.admin.create(newPermissionDto); 
```